modules/index-rebuild/src/main/java/org/opencastproject/index/rebuild/AbstractIndexProducer.java

```diff
@@ -0,0 +1,215 @@
+   * @param org
+   *           The organization.
+   */
+  protected void logIndexRebuildBegin(String indexName, int total, String elementName, Organization org) {
+    logger.info("Starting update of index '{}' for service '{}' with {} {} of organization '{}'", indexName, getService(),
+            total, elementName, org);
+  }
+
+  /**
+   * Log the progress of the index rebuild for this service.
+   *
+   * @param indexName
+   *           The name of the index that's being rebuild.
+   * @param total
+   *           The total amount of elements to be re-added.
+   * @param current
+   *           The amount of elements that have already been re-added.
+   */
+  protected void logIndexRebuildProgress(String indexName, int total, int current) {
+    logIndexRebuildProgress(indexName, total, current, 1);
+  }
+
+  /**
+   * Log the progress of the index rebuild for this service and a specific organization.
+   *
+   * @param indexName
+   *           The name of the index that's being rebuild.
+   * @param total
+   *           The total amount of elements to be re-added.
+   * @param current
+   *           The amount of elements that have already been re-added.
+   * @param org
+   *           The organization.
+   */
+  protected void logIndexRebuildProgress(String indexName, int total, int current, Organization org) {
+    logIndexRebuildProgress(indexName, total, current, 1, org);
+  }
+
+  /**
+   * Log the progress of the index rebuild for this service.
+   *
+   * @param indexName
+   *           The name of the index that's being rebuild.
+   * @param total
+   *           The total amount of elements to be re-added.
+   * @param current
+   *           The amount of elements that have already been re-added.
+   * @param batchSize
+   *           The size of the batch we re-add in one go.
+   */
+  protected void logIndexRebuildProgress(String indexName, int total, int current, int batchSize) {
+    logIndexRebuildProgress(indexName, total, current, batchSize, null);
+  }
+
+  /**
+   * Log the progress of the index rebuild for this service.
+   *
+   * @param indexName
+   *           The name of the index that's being rebuild.
+   * @param total
+   *           The total amount of elements to be re-added.
+   * @param current
+   *           The amount of elements that have already been re-added.
+   * @param batchSize
+   *           The size of the batch we re-add in one go.
+   * @param org
+   *           The organization (can be null).
+   */
+  protected void logIndexRebuildProgress(String indexName, int total, int current, int batchSize, Organization org) {
+    final int responseInterval = (total < 100) ? 1 : (total / 100);
+    if (responseInterval == 1 || batchSize > responseInterval || current == total
+            || current % responseInterval < batchSize) {
+
+      if (org == null) {
+        logger.info("Updating index '{}' for service '{}': {}/{} finished, {}% complete.", indexName, getService(),
```

Comment:
Would it make sense to pass the logger along as well to not loose context information?
(A question for most of the methods in there)

Reply:
I did not think of that. Good idea. :D